### ISSUE TITLE
Primary key with value 0

### ISSUE THREAD

**Issue by simara-esports** [nette/database#13]
Hi,

if there is a row with value "0" in primary key column, the field won't appear in the result.
It is overwritten (or can overwrite other row).
The problem is here: https://github.com/nette/database/blob/master/src/Database/Table/Selection.php#L506
quickfix:
```php
$this->rows[($primary || $primary === "0" || $primary === 0) ? $primary : $key] = $row;
```

But i'm not sure how clear is this solution.

**Comment by mdjimy** [comment]
Yes, the solution you have proposed should work just fine (to solve yours problem). Maybe little bit modified to:

````php
...
$primary = $row->getSignature(FALSE);
$usedPrimary &= $primary || $primary === '0';
$this->rows[$primary || $primary === '0' ? $primary : $key] = $row;
...
````

But, I've found out, that problem is more complex, than it may seem!
Lets work with a table which have composite (multi-column) primary key, where atleast one of primary columns is of type datetime, date, time, timestamp or interval.

Then an exception is thrown due to fact, that method getSignature tries to convert given DateTime or DateInterval object to string, but those objects don't have __toString method.

I have already prepared bug fix, but before I'll create pull request I have to write some test cases :)
